### ISSUE TITLE
JavaScript/Recursive Methods: "Visualizing Recursion in the call stack" resource now behind paywall

### ISSUE THREAD

**Issue by scottwright-dev** [TheOdinProject/curriculum#26622]
### Describe your suggestion

Having checked the additional resources section in the Recursive Methods lesson I can report that the "Visualizing Recursion in the call stack "appears to have now been moved behind a paywall for members only. 

I have double checked this with my (free tier) Medium account and cannot access the full article. I think it may have to be removed as a resource.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-recursive-methods

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

**Comment by CouchofTomato** [comment]
Agreed. Options are to find another resource or simply remove this one.

If you don't have another resource in mind then just submit a PR removing the one there. I'll assign it to you since you have indicated you want to work on this issue.

**Comment by scottwright-dev** [comment]
Thanks for assigning I can certainly remove the existing link. As a consideration for a replacement, as the main focus of the article appears to provide the learner with a way of visualising how a recursive function works I have a suggestion of an alternative resource which could forfil this. 

There is a site I have used in the past which I find helpful in visualising whats happening in the code: [Python Tutor](https://pythontutor.com/), the site has a helpful visualiser tool where you can input code in Javascript and it allows you to step through the sequence of the function. 

For demonstration purposes, I plugged in one of the questions from the assignment of this lesson, [example here](https://pythontutor.com/visualize.html#code=//%20This%20function%20calculates%20the%20factorial%20of%20a%20given%20number%20using%20recursion.%0Afunction%20factorial%28num%29%20%7B%0A%20%20%20%20//%20Base%20case%3A%20If%20the%20number%20is%201,%20return%201.%0A%20%20%20%20//%20This%20is%20the%20point%20where%20the%20recursive%20calls%20will%20stop.%0A%20%20%20%20if%20%28num%20%3D%3D%3D%201%29%20return%201%3B%0A%0A%20%20%20%20//%20Recursive%20case%3A%20Multiply%20the%20current%20number%20by%20the%20factorial%20of%20%28number%20-%201%29.%0A%20%20%20%20//%20This%20means%20the%20function%20will%20keep%20calling%20itself%20with%20decreasing%20values%20of%20'num'%20until%20it%20reaches%201.%0A%20%20%20%20return%20num%20*%20factorial%28num%20-%201%29%3B%0A%7D%0A%0A//%20Calling%20the%20factorial%20function%20with%20the%20value%20of%203.%0A//%20The%20sequence%20of%20operations%20is%3A%203%20*%20%282%20*%20%281%29%29%20%3D%206%0Afactorial%283%29%20//%20answer%20is%3A%206%0A&cumulative=false&heapPrimitives=nevernest&mode=edit&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false). 

Perhaps this could be something to consider? I would be interested to hear your thoughts if you feel something like this tool brings some additional value to the lesson.  If not I can go ahead and remove the existing resource.

**Comment by scottwright-dev** [comment]
Sure, sounds like a plan. For clarity would an example similar to the one used in the original article suffice? 
The article used factorials. So something along the lines of:

```javascript
function calcFactorial(num) {
    if (num === 1) {
        return 1;
    }
    return num * calcFactorial(num - 1);
}

calcFactorial(5);
```

Which would be linked like [this](https://pythontutor.com/render.html#code=function%20calcFactorial%28num%29%20%7B%0A%20%20%20%20if%20%28num%20%3D%3D%3D%201%29%20%7B%0A%20%20%20%20%20%20%20%20return%201%3B%0A%20%20%20%20%7D%0A%20%20%20%20return%20num%20*%20calcFactorial%28num%20-%201%29%3B%0A%7D%0A%0AcalcFactorial%285%29%3B&cumulative=false&curInstr=0&heapPrimitives=nevernest&mode=display&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false) in the visualizer. Please let me know if thats suitable and I'll add it in. Many Thanks.

**Comment by Baguirre03** [comment]
Nooo. Thats so devastating, I wonder why on earth they would change that. I agree, maybe the visualizer would be nice, or maybe I could write something similar up 🤔

**Comment by scottwright-dev** [comment]
Hi @CouchofTomato, I've submitted a PR with the replacement of the resource as we discussed. 

The Python Tutor visualizer I've included is reasonably intuitive I think, but I'm mindful that new users might benefit from some additional guidance. Please let me know if you feel we should add a brief instruction on how to use the tool.

Many Thanks,

Scott
